tests/protocol/mocks.py

```diff
@@ -27,13 +27,15 @@
 
 
 @contextlib.contextmanager
-def mock_wire_protocol(mock_wire_data_file, http_get_handler=None, http_post_handler=None, http_put_handler=None, fail_on_unknown_request=True):
+def mock_wire_protocol(mock_wire_data_source, http_get_handler=None, http_post_handler=None, http_put_handler=None, fail_on_unknown_request=True,
```

Comment:
it's odd to take both mock_wire_data_source and mockwiredata_factory

it seems like the expectation of the code is that when the source is a file the factory must be aget_file_based_wire_protocol_data and when a fetcher it must be get_dynamic_wire_protocol_data

let's try to improve this design